containers/entity-resolution/handlers/entityResolution.go

```diff
@@ -0,0 +1,237 @@
+type EntityResolution struct {
+	Identifier        string   `json:"identifier" example:"bob@sample.org"`
+	EntityIdentifiers []string `json:"entity_ids" example:"XUX-ASDF-ASDFSD"`
+}
+
+// EntityResolution Response Model
+type EntityResolutionResponse struct {
+	AttributeType     string             `json:"type" example:"email"`
+	EntityResolutions []EntityResolution `json:"entity_resolutions"`
+}
+
+// EntityResolution request model
+type EntityResolutionRequest struct {
+	//enum: email,username
+	AttributeType     string   `json:"type" example:"email"`
+	EntityIdentifiers []string `json:"identifiers" example:"bob@sample.org"`
+}
+
+type KeyCloakConnector struct {
+	token  *gocloak.JWT
+	client gocloak.GoCloak
+}
+
+type KeyCloakConfg struct {
+	Url          string
+	Realm        string
+	ClientId     string
+	ClientSecret string
+	AuthPath     bool `default:"true"`
+	SubGroups    bool `default:"false"`
+}
+
+// GetEntitlements godoc
+// @Summary      Resolve a set of entity labels to their keycloak identifiers
+// @Description  Provide an attribute type and attribute label list
+// @Description  and receive a list of entity idenitifiers
+// @Success      200 {object} handlers.EntityResolutionResponse
+// @Param        "Entity Resolution Request" body handlers.EntityResolutionRequest true "Entity Identifiers to be resolved"
+// @Accept       json
+// @Produce      json
+// @Router       /resolve [post]
+func GetEntityResolutionHandler(kcConfig KeyCloakConfg, logger *zap.SugaredLogger) http.Handler {
+	//Try initial login
+	c, err := getKCClient(kcConfig, logger)
+	if err != nil {
+		logger.Fatal("Error connecting to keycloak")
+	}
+	logger.Debug("client token", c.token.AccessToken)
+
+	ctxb := ctx.Background()
+
+	entityResolutionHandler := func(w http.ResponseWriter, req *http.Request) {
+		spanCtx := req.Context()
+		handlerCtx, span := tracer.Start(spanCtx, "GetEntityResolutionHandler")
+		defer span.End()
+
+		if req.Method != http.MethodPost {
+			logger.Error("Rejected request, invalid HTTP verb")
+			http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
+			return
+		}
+
+		logger.Debug("GetEntityResolutionHandler - reading request body")
+
+		//Read + consume body
+		bodBytes, err := ioutil.ReadAll(req.Body)
+		defer req.Body.Close()
+		if err != nil {
+			logger.Errorf("Couldn't read client request! Error was %s", err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		payload, err := getRequestPayload(bodBytes, handlerCtx, logger)
+		if err != nil {
+			logger.Errorf("Couldn't deserialize client request! Error was %s", err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		var resolvedEntities []EntityResolution
+		logger.Debugf("Received payload", payload)
+
+		kcConnector, err := getKCClient(kcConfig, logger)
+		if err != nil {
+			logger.Errorf("Error connecting to keycloak %s", err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		for i := 0; i < len(payload.EntityIdentifiers); i++ {
```

Comment:
```suggestion
		for idx, ident := range payload.EntityIdentifiers
```

`range` would be more appropriate here

Reply:
Fixed